argparse/templates/cwltool_inputs.j2

```diff
@@ -0,0 +1,40 @@
+{% extends "cwltool.j2" %}
+
+{% block cwltool_inputs %}
+{% for param in tool.inputs %}
+
+- id: {{ param.id }}
+  type: {% if param.type != "array" %}{{ param.type }}{% else %}
+
+    type: {{ param.type }}
+    items: {{ param.items_type or "string" }}{% endif -%}
+
+  {% if param.default %}
+
+  default: {{ param.default }}{% endif %}
+
+  {% if param.description %}  description: {{ param.description }}{% endif %}
+
+  inputBinding:
+    position: {{ param.position }}
+{% endfor %}
+- id: {{ tool.name }}
+  type: File
+  default:
+    class: File
+    path: {{ tool.name }}
+  inputBinding:
```

Comment:
I guess this template will need further factorization for better composability. I guess that a generic `id.j2` template would help with positional arguments auto-numbering (i.e: `position 0`, `position 1`, etc...)? I'm sure you already thought about that, just thinking out loud here.

@mr-c